File: python/src/problem/leetcode/easy/leetcode_26.py
# coding=utf-8
from typing import List


class Solution:

    @staticmethod
    def remove_duplicates(nums: List[int]) -> int:
        """
        Time: O(n), Space: O(1)
        :param nums:
        :return:
        """
        if not nums:
            return 0
        right = left = 1
        while right < len(nums):
            if nums[right] != nums[right - 1]:
                nums[left] = nums[right]
                left += 1
            right += 1
        return left

File: c/easy/leetcode_26.c
//
// Created by ywh on 2020/8/15.
//

/**
 * 删除排序数组中的重复项
 *
 * @param nums
 * @param numsSize
 * @return
 */
int removeDuplicates(int *nums, int numsSize) {
    if (!nums || numsSize <= 0) {
        return 0;
    }
    int left = 1, right = 1;
    for (; right < numsSize; right++) {
        if (nums[right] != nums[right - 1]) {
            nums[left++] = nums[right];
        }
    }
    return left;
}